### PR TITLE
Release notes list design implementation

### PR DESCRIPTION
**Pull request by davifantasia**
Implementation of this [design](https://commercetools.invisionapp.com/d/main?origin=v7#/console/19438556/408192223/preview?scrollOffset=215)

### CODE REVIEW

File: packages/gatsby-theme-docs/src/layouts/internals/layout-release-note.js [hunk 1]
```diff
@@ -1,27 +1,78 @@
...
+`;
+const Type = styled.p`
+  display: inline-block;
+  padding: ${designSystem.dimensions.spacings.xs}
+    ${designSystem.dimensions.spacings.s};
+  font-size: 12px;
+  line-height: 16px;
+  border: 1px solid #00ccb4;
+  border-radius: 2px;
+  background-color: #b3fff6;
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-release-note.js:20]: No magic numbers please. All those values should be defined in the design system, and I presume most of them already are.

File: packages/gatsby-theme-docs/src/layouts/internals/layout-release-note.js [hunk 2]
```diff
@@ -1,27 +1,78 @@
...
-      <ReleaseNoteTitle>{props.title}</ReleaseNoteTitle>
-      <div>{props.date}</div>
+      <ReleaseNoteTitle>{props.releaseNote.title}</ReleaseNoteTitle>
+      <DateElement>{props.releaseNote.date}</DateElement>
+      <div>
+        <Type>{props.releaseNote.type.toLowerCase()}</Type>
+      </div>
+      {props.releaseNote.topics.length > 0 && (
+        <div>
+          <table>
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-release-note.js:48]: Sorry but let's use css-grid or flexbox here please. There is no need to use tables here.

File: packages/gatsby-theme-docs/src/layouts/internals/layout-release-note.js [hunk 3]
```diff
@@ -1,27 +1,78 @@
...
+        <MDXRenderer>{props.releaseNote.body}</MDXRenderer>
       </div>
     </SpacingsStack>
   );
 };
 ReleaseNote.propTypes = {
-  title: PropTypes.string.isRequired,
-  date: PropTypes.string.isRequired,
-  body: PropTypes.string.isRequired,
+  releaseNote: PropTypes.shape({
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-release-note.js:67]: This prop is redundant. Just spread the values

```js
<LayoutReleaseNote {...releaseNote} />
```

File: packages/gatsby-theme-docs/src/layouts/internals/layout-release-note.js [hunk 4]
```diff
@@ -1,27 +1,106 @@
...
+
+const featureStyle = css`
+  background-color: ${designSystem.colors.light
+    .surfaceReleaseNoteNewFeatureType};
+  border-color: ${designSystem.colors.light.borderReleaseNoteNewFeatureType};
+`;
+const enhancementStyle = css`
+  background-color: ${designSystem.colors.light
+    .surfaceReleaseNoteEnhancementType};
+  border-color: ${designSystem.colors.light.textInfo};
```
[the remark this answers is not in the document]
Comment (davifantasia) [reply packages/gatsby-theme-docs/src/layouts/internals/layout-release-note.js:70]: If textInfo is `#078CDF` then that is correct.

File: packages/gatsby-theme-docs/src/layouts/internals/layout-release-note.js [hunk 5]
```diff
@@ -1,27 +1,106 @@
...
+  font-size: ${designSystem.typography.fontSizes.small};
+  span {
+    padding: ${designSystem.dimensions.spacings.xs};
+  }
+  span:first-of-type {
+    padding-left: 0;
+  }
+  span:not(:first-of-type) {
+    border-left: 1px solid ${designSystem.colors.light.surfaceSecondary3};
+  }
```
[the remark this answers is not in the document]
Comment (davifantasia) [reply packages/gatsby-theme-docs/src/layouts/internals/layout-release-note.js:26]: I don't know how to use this selector.

```
span > * + * {}
```

[the remark this answers is not in the document]
Comment (emmenko) [reply packages/gatsby-theme-docs/src/layouts/internals/layout-release-note.js:26]: Have you google it? It basically selects the children except the first one.

File: packages/gatsby-theme-docs/src/layouts/internals/layout-release-note.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (emmenko) [reply packages/gatsby-theme-docs/src/layouts/internals/layout-release-note.js:71]: Of course, my mistake sorry. I read this as `textPrimary` 🤦‍♂️

**Comment by davifantasia** [comment]
> There is still the prop type to be fixed

Where exactly? Please comment on the line again.

File: packages/gatsby-theme-docs/src/templates/release-notes-list.js [hunk 6]
```diff
@@ -100,7 +95,7 @@ export const query = graphql`
       nodes {
         slug
         title
-        date(formatString: "YYYY-MM-DD")
+        date(formatString: "D. MMMM YYYY")
```
[the remark this answers is not in the document]
Comment (nkuehn) [reply packages/gatsby-theme-docs/src/templates/release-notes-list.js:98]: Nearly,  remote the dot and we're fine  (I'll improve the writing style checker's hint text which is not up to date).

[the remark this answers is not in the document]
Comment (emmenko) [reply packages/gatsby-theme-docs/src/templates/release-notes-list.js:98]: I mean, are we using the english format or how is it supposed to be?

**Comment by davifantasia** [comment]
@emmenko could you please take a look at the checks in this PR? They are not completing.

**Comment by emmenko** [comment]
All good 🙌

**Comment by davifantasia** [comment]
> All good 🙌

Thanks :)

**Pull request merged**